modules/Bio/EnsEMBL/Production/Pipeline/Common/SpeciesFactory.pm

```diff
@@ -169,7 +169,19 @@ sub has_variation {
   my $dbva =
     Bio::EnsEMBL::Registry->get_DBAdaptor( $species, 'variation' );
 
-  my $has_variation = defined $dbva ? 1 : 0;
+  my $has_variation;
```

Comment:
I'm not convinced that this is the best approach. In the context of the variation stats pipeline this works, because we don't want to do anything with a variation db that has no variants in it. But in other cases, that might not be true - for example, the FTPChecker pipeline uses the variation dataflow in order to determine whether variants should exists for a species. And you could imagine other use cases, e.g. dumping all databases associated with a species, for which this would not dump the variation db, which is properly not what you expect/want.
My suggestion would be to turn the 'VariationTasks' from a dummy module to a proper one, with this source logic inside it, controlling whether anything is flowed.

Reply:
@james-monkeyshines: That is a very good point but the issue is that many pipelines are using Species variation factory including mart pipelines and all these pipelines expect a "complete" variation database. As we have just discovered, they don't generate any dumps for these databases so it should be fine for FTPChecker and it should not check these dbs. For database dumping, we are getting the list of database from the metadata database at the moment. The Metadata pipeline don't process these variation dbs since they don't have any data in it.
I believe it is better to change the species factory than all the different pipelines using it and this would avoid pipelines to die because they can't process these dbs. 
I think Variation should have consulted other teams about the possible side effect of these databases instead of pushing this live.
Alternatively, we could always have a new variation flow with the variation vcf dbs in it so that they are not lost. I think that if variation wants us to process these databases in the future then we would need to implement something new and possibly update our pipelines.

Reply:
OK, yes, I agree with your argument. Having a separate flow for vcf-only dbs makes sense, but I don't think we need to bother implementing it until we have a use case.